mlog/log.go

```diff
@@ -190,9 +193,9 @@ func (l *Logger) StdLogWriter() io.Writer {
 }
 
 func (l *Logger) WithCallerSkip(skip int) *Logger {
-	newlogger := *l
-	newlogger.zap = newlogger.zap.WithOptions(zap.AddCallerSkip(skip))
-	return &newlogger
+	newLogger := *l
+	newLogger.zap = newLogger.zap.WithOptions(zap.AddCallerSkip(skip))
```

Comment:
Could this trigger similar issues?

Reply:
hmmmm...I think this is fine since it operates on the main logger struct and not on the `logger.Logr` right?

Reply:
Looks fine since it's done on a newly created object and from that point on it should be all reads.
Even `StdLogWriter` does a copy before setting that so we are good.

Reply:
Exactly!!!